### PR TITLE
Add transportserver-template config

### PR DESCRIPTION
**Pull request by ssrahul96**
### Proposed changes
Created custom-template Config for `TransportServer`

Fixes https://github.com/nginxinc/kubernetes-ingress/issues/5459

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [ ] I have rebased my branch onto main
- [ ] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by ADubhlaoich — COMMENTED** [review]
LGTM docs-wise but the majority of the PR is code, so I will not approve.

**Comment by ssrahul96** [comment]
Team, Could it be reviewed and merged?

**Comment by j1m-ryan** [comment]
Hi @ssrahul96 , can you create an issue for this, and attach it to this PR?

File: examples/shared-examples/custom-templates/README.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ via the following keys:
 - `main-template` - Sets the main NGINX configuration template.
 - `ingress-template` - Sets the Ingress NGINX configuration template for an Ingress resource.
 - `virtualserver-template` - Sets the NGINX configuration template for an VirtualServer resource.
+- `trasnportserver-template` - Sets the NGINX configuration template for an TransportServer resource.
```
Comment (brianehlert) [inline examples/shared-examples/custom-templates/README.md:10]: misspelling

File: examples/shared-examples/custom-templates/README.md [hunk 2]
```diff
@@ -35,6 +36,12 @@ data:
     ...
     }
     {{ end }}
+  trasnportserver-template: |
```
Comment (brianehlert) [inline examples/shared-examples/custom-templates/README.md:39]: misspelling repeat

File: docs/content/configuration/global-configuration/configmap-resource.md [hunk 3]
```diff
@@ -173,6 +173,7 @@ See the doc about [VirtualServer and VirtualServerRoute resources](/nginx-ingres
 |``main-template`` | Sets the main NGINX configuration template. | By default the template is read from the file in the container. | [Custom Templates](/nginx-ingress-controller/configuration/global-configuration/custom-templates). |
 |``ingress-template`` | Sets the NGINX configuration template for an Ingress resource. | By default the template is read from the file on the container. | [Custom Templates](/nginx-ingress-controller/configuration/global-configuration/custom-templates). |
 |``virtualserver-template`` | Sets the NGINX configuration template for an VirtualServer resource. | By default the template is read from the file on the container. | [Custom Templates](/nginx-ingress-controller/configuration/global-configuration/custom-templates). |
+|``transportserver-template`` | Sets the NGINX configuration template for an TransportServer resource. | By default the template is read from the file on the container. | [Custom Templates](/nginx-ingress-controller/configuration/global-configuration/custom-templates). |
```
Comment (j1m-ryan) [inline docs/content/configuration/global-configuration/configmap-resource.md:176]:
```suggestion
|``transportserver-template`` | Sets the NGINX configuration template for a TransportServer resource. | By default the template is read from the file on the container. | [Custom Templates](/nginx-ingress-controller/configuration/global-configuration/custom-templates). |
```

File: examples/shared-examples/custom-templates/README.md [hunk 4]
```diff
@@ -7,6 +7,7 @@ via the following keys:
 - `main-template` - Sets the main NGINX configuration template.
 - `ingress-template` - Sets the Ingress NGINX configuration template for an Ingress resource.
 - `virtualserver-template` - Sets the NGINX configuration template for an VirtualServer resource.
+- `transportserver-template` - Sets the NGINX configuration template for an TransportServer resource.
```
Comment (j1m-ryan) [inline examples/shared-examples/custom-templates/README.md:10]:
```suggestion
- `transportserver-template` - Sets the NGINX configuration template for a TransportServer resource.
```

**Review by j1m-ryan — APPROVED** [review]

**Review by AlexFenlon — APPROVED** [review]

**Pull request merged**